samples/cors/ts/README.md

```diff
@@ -0,0 +1,105 @@
+
+If your web application already has a back end component this will be the most straightforward approach. You'll simply add a new API endpoint to your server which will be responsible for fetching secrets from Azure Key Vault and returning it to the user. This allows you to have complete control and customization over the security and authorization of calls to Key Vault since all calls will go through a priviledged environment (your server).
+
+## Use Azure API Management
+
+But what if you don't have a server in place for your Single Page Application? Thankfully Azure provides a managed API service that can sit between your web application and Azure Key Vault and provide the CORS policy that you need to interact with Azure Key Vault.
+
+It provides all of the same benefits as having your own back end API while avoiding the need to separately deploy, manage, and secure your own server. You can define a single API that is responsible for retrieving secrets and apply best-in-class security to control access to your Key Vault.
+
+# Sample code
+
+This simple example demonstrates how to get started with both of these approaches. In this example, we create a simple [Express][express] application, connect and upload a single test secret, and fetch it via a web application using either our back end server or Azure API Management.
+
+## Security considerations
+
+This sample demonstrates a few alternatives to integrating with Azure Key Vault from the browser, and is purposely kept simple. Remember, the browser is an _insecure_ environment and we encourage you to familiarize yourself with Azure's security policies to avoid leaking credentials to an unauthorized user in a production application. Please refer to Azure Key Vault's [security overview][keyvaultsecurity] to learn more about securing your Key Vault's data.
+
+## Prerequisites
+
+The sample is compatible with Node.js >= 8.0.0
+
+Before running the samples in Node, they must be compiled to JavaScript using the TypeScript compiler. For more information on TypeScript, see the [TypeScript documentation][typescript].
+
+You need [an Azure subscription][freesub] and the following resources created to run this sample:
+
+- An Azure Key Vault. Please refer to the [Key Vault documentation][keyvault] for additional information on Azure Key Vault.
+- An Azure API Management. Please refer to [Azure API Management][azureapimanagement] for additional information on Azure API Management.
+
+To quickly create the necessary resources in Azure and to receive the necessary environment variables for them, you can deploy our sample template by clicking:
+
+[![](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-sdk-for-js%2Fmaster%2Fsamples%2Fcors%2Farm-template.json)
```

Comment:
I seem to recall that ARM has two deployment modes - incremental and <something else>. I believe the non-incremental one is destructive. Do you know if your template warns them when they deploy against an existing keyvault (since you are modifying the keyvault?).

Reply:
(there's a note in here about it, but it's under the incremental section)

https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-modes#incremental-mode

Reply:
Thanks for that! I read in that doc that incremental mode is the default - so I _think_ this should be ok? Unless that behavior is different when deploying through the portal - do you happen to know? I can experiment if you don't know off the top of your head.

Reply:
Ah, scratch this feedback. I misread your template and thought you were asking them to specify a specific resource - it looks like you're generating one and the intention is to create a new one anyways. So no big deal.

It is still an interesting thing about ARM though, so hopefully not a total waste of your time! ;)